### PR TITLE
Adding Mac & Windows to build

### PR DESCRIPTION
**Pull request by MWSestabro**
Updated build.yml to include Mac and Windows

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -33,3 +33,56 @@ jobs:
...
+            OPENTELEMETRY_MATLAB_INSTALL: "${{ github.workspace }}/otel_matlab_install"
+        steps:
+            - name: Download OpenTelemetry-Matlab source
+              uses: actions/checkout@v3
+              with: 
+                path: opentelemetry-matlab
+            - name: Install MATLAB
+              uses: matlab-actions/setup-matlab@v1
+              with: 
+                release: R2023a
```
Comment (duncanpo) [inline .github/workflows/build.yml:48]: Is it necessary to pin MATLAB to a specific release? Or can we just let it use the latest?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -33,3 +33,56 @@ jobs:
...
+            - name: Build OpenTelemetry-Matlab
+              run: |
+                  cd opentelemetry-matlab
+                  cmake --version
+                  cmake -S . -B build -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=${{ env.OPENTELEMETRY_MATLAB_INSTALL }}
```
Comment (duncanpo) [inline .github/workflows/build.yml:53]: I would add -DWITH_EXAMPLES=ON. Even though the tests currently don't test the examples, it is good to at least build them and make sure they build successfully

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -33,3 +33,56 @@ jobs:
...
+                  cmake --build build --config Release --target install -v
```
Comment (duncanpo) [inline .github/workflows/build.yml:54]: The verbose flag -v is probably not necessary?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -33,3 +33,56 @@ jobs:
...
+            OPENTELEMETRY_MATLAB_INSTALL: "${{ github.workspace }}/otel_matlab_install"
+        steps:
+            - name: Download OpenTelemetry-Matlab source
+              uses: actions/checkout@v3
+              with: 
+                path: opentelemetry-matlab
+            - name: Install MATLAB
+              uses: matlab-actions/setup-matlab@v1
+              with: 
+                release: R2023a
```
Comment (duncanpo) [inline .github/workflows/build.yml:75]: Again is pinning to a specific release necessary?

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -33,3 +33,56 @@ jobs:
...
+            - name: Build OpenTelemetry-Matlab
+              run: |
+                  cd opentelemetry-matlab
+                  cmake -S . -B build -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=${{ env.OPENTELEMETRY_MATLAB_INSTALL }}
```
Comment (duncanpo) [inline .github/workflows/build.yml:79]: Add -DWITH_EXAMPLES=ON.

**Review by duncanpo — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (MWSestabro) [reply .github/workflows/build.yml:54]: -DWITH_EXAMPLES=ON has been added on Linux & Mac. Windows is currently experiencing failures Duncan is investigating.

File: .github/workflows/build.yml (hunk 3 above)
Comment (MWSestabro) [reply .github/workflows/build.yml]: Verbose flag removed

File: .github/workflows/build.yml (hunk 4 above)
Comment (MWSestabro) [reply .github/workflows/build.yml]: The MATLAB release has been removed from the build on all platforms.

File: .github/workflows/build.yml (hunk 5 above)
Comment (MWSestabro) [reply .github/workflows/build.yml]: -DWITH_EXAMPLES=ON has been added on Linux & Mac. Windows is currently experiencing failures Duncan is investigating.

**Review by duncanpo — APPROVED** [review]

**Pull request merged**